server.js

```diff
@@ -51,37 +61,72 @@ let messageList = [];
 const addToList = message => messageList.push(message);
 subscribers.push(addToList);
 
-// https://expressjs.com/en/starter/basic-routing.html
-app.get("/", (request, response) => {
-  response.sendFile(__dirname + "/views/index.html");
-});
+const routeGetRootWithUserId = config => (request, response) => {
+  let template = PageWrapper({
+    title: config.pageTitle,
+    children: ChatRoom({
+      userName: request.params.id ? request.params.id : config.defaultUserName
+    })
+  });
+  response.send(template);
+}
 
-app.get("/messages", (request, response) => {
-  response.type("text/plain");
+const routeGetRoot = config => (request, response) => {
+  let template = PageWrapper({
+    title: config.pageTitle,
+    children: SignIn()
+  });
+  response.send(template);
+}
+
+const routePostRoot = config => (request, response) => {
+  response.redirect(`/${request.body.user}`);
+}
+
+const routeGetMessages = config => (request, response) => {
+  response.type('text/html');
   const write = sentence => {
-    const line = `${sentence}\n`;
+    const line = `${sentence}`;
     response.write(line);
   };
-
-  subscribers.push(write);
   const cutoff = Math.max(messageList.length - 10, 0);
-
   const latest = messageList.slice(cutoff);
   if (cutoff !== 0) {
     messageList = latest;
   }
-
+  subscribers.push(write);
   messageList.forEach(write);
-});
+};
 
-app.post("/messages", (request, response) => {
+const routePostMessages = config => (request, response) => {
   if (request.body.message) {
-    publish(`me: ${request.body.message}`);
+    publish(ChatText({
+      color: stringToColor(request.body.user),
+      user: request.body.user,
+      children: request.body.message
+    }));
   }
-  response.redirect("/");
-});
+  response.redirect(`/${request.body.user}`);
+}
+
+// this should come from a json file someday...
+const appConfig = {
+  listenPort: 3301,
+  pageTitle: 'This is a chatroom',
+  defaultUserName: 'Moose'
+}
+
+const config = {};
+// Routes
+app.get("/messages", routeGetMessages(appConfig));
+app.post("/messages", routePostMessages(appConfig));
+
+app.get("/", routeGetRoot(appConfig));
+app.post("/", routePostRoot(appConfig));
```

Comment:
this is the post from the index page.
All this does is setup the chat page form/url with a user parameter that we can key off of to make a "unique" representation of a user.

A future improvement would be to get a list of subscribers by name and check tomake sure we don't have duplicates?